source/gui/settingsDialogs.py

```diff
@@ -5520,3 +5576,14 @@ def onFilterEditTextChange(self, evt):
 		self.filter(self.filterEdit.Value)
 		self._refreshVisibleItems()
 		evt.Skip()
+
+
+def _isResponseAddonStoreCacheHash(response: requests.Response) -> bool:
+	try:
+		# Attempt to parse the response as JSON
+		data = response.json()
+	except ValueError:
+		# Add-on Store cache hash is JSON, so this can't be it.
+		return False
+	# Add-on Store cache hash is a git commit hash as a string.
+	return isinstance(data, str) and bool(re.fullmatch("[0-9a-fA-F]{7,40}", data))
```

Comment:
ah I don't think this has to be true - we use a commit hash but mirrors may want to use their own cache hash, like a new UUID every X minutes to cache break, or whenever they generate new data.

I would just ensure there is a non-empty string

Reply:
```suggestion
	# Add-on Store cache hash must be a non-empty JSON string
	return isinstance(data, str) and bool(re.fullmatch(".+", data))
```

Reply:
I've updated this to check whether the string is not the empty string, but I have not used your suggested code as I believe using `re` here is overkill. I've just used `bool(data)` instead. Let me know if I've missed some reason this won't work as expected.